File: server/Models/FFRanks.cs
using MongoDB.Bson.Serialization.Attributes;

namespace server.Models;

public class FFRanking
{
    [BsonId]
    [BsonRepresentation(MongoDB.Bson.BsonType.ObjectId)]
    public string Id { get; set; } = string.Empty;

    [BsonElement("participantName")]
    public string ParticipantsName { get; set; } = string.Empty;

    [BsonElement("RankingList")]
    public List<ShowRanks> RankingList { get; set; } = new();
}

public class ShowRanks
{
    [BsonElement("showId")]
    public string ShowId { get; set; } = string.Empty;

    [BsonElement("Rank")]
    public int Rank { get; set; }
}

public class RankingResult
{
    [BsonElement("showId")]
    public string ShowId { get; set; } = string.Empty;

    [BsonElement("Points")]
    public int Points { get; set; }
}

File: app/utils/types/ffranks.ts
export type ShowRanks = {
  showId: string;
  rank: number;
};

export type FFRanking = {
  id: string;
  participantsName: string;
  rankingList: ShowRanks[];
};

export type RankingResult = {
  showId: string;
  points: number;
};

export type FFPointsDTO = {
  showId: string;
  showName: string;
  showImage: string;
  points: number;
};

export type FFRankDTO = {
  showId: string;
  showName: string;
  showImage: string;
  rank: number;
  tvMazeId: number;
};

export type AdminRankingRequest = {
  name?: string;
  tvmazeIds: number[];
};


export type UserRankingRequest = {
  name: string;
  tvmazeIds: number[];
};

export type FFParticipantRanking = {
  name: string;
  rankings: FFRankDTO[];
};